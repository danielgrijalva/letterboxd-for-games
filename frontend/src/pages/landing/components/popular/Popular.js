import React from "react";
import './Popular.css'

export const Popular = ({isLoading, popular}) => (
  <section className="popular margin-top-xs margin-bottom">
    {!isLoading ? (
      <React.Fragment>
        {popular.map(p => {
          return (
            <div className="cover-wrapper">
              <img
                key={p}
                className="cover"
                src={`https://images.igdb.com/igdb/image/upload/t_cover_big/${
                  p.image_id
                }.jpg`}
              />
              <div className="cover-overlay">
                <strong>{p.name}</strong>
              </div>
            </div>
          );
        })}
      </React.Fragment>
    ) : (
      <React.Fragment>
        {[...Array(6)].map(i => (
          <div key={i} className="placeholder" />
        ))}
      </React.Fragment>
    )}
  </section>
);
