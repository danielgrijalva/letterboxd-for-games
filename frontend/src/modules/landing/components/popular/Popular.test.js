import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Popular from '../popular';

describe('Test <Popular />', () => {

  it('shows required props warning when no props passed', () => {
    console.error = jest.fn();
    expect(() => {
      shallow(<Popular />);
    }).toThrow();

    expect(console.error).toHaveBeenCalledTimes(2);
    const isLoadingWarning = console.error.mock.calls[0];
    const popularWarning = console.error.mock.calls[1];
    expect(isLoadingWarning[0]).toMatch("The prop `isLoading` is marked as required in `Popular`,");
    expect(popularWarning[0]).toMatch("The prop `popular` is marked as required in `Popular`,");
    console.error.mockClear();
  });

});

describe('Test <Popular /> when isLoading is true', () => {

  let wrap;
  const props = {
    isLoading: true,
    popular: [],
  }

  beforeAll(() => {
    wrap = shallow(<Popular {...props}/>);
  });

  it('renders correctly and matches snapshot', () => {
    expect(toJson(wrap)).toMatchSnapshot();
  })

  it('renders a section with popular margin-top-xs margin-bottom classes', () => {
    const section = wrap.find('section');
    expect(section.hasClass('popular margin-top-xs margin-bottom')).toEqual(true);
  });

  it('renders 6 placeholders', () => {
    const placeholders = wrap.find('div');
    expect(placeholders).toHaveLength(6);
    placeholders.forEach((node) => {
      expect(node.hasClass('placeholder')).toEqual(true);
    });
  });
});
