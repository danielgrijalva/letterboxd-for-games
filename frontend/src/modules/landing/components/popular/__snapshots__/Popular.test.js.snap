// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Popular /> when isLoading is true renders correctly and matches snapshot 1`] = `
<section
  className="popular margin-top-xs margin-bottom"
>
  <div
    className="placeholder"
    key="0"
  />
  <div
    className="placeholder"
    key="1"
  />
  <div
    className="placeholder"
    key="2"
  />
  <div
    className="placeholder"
    key="3"
  />
  <div
    className="placeholder"
    key="4"
  />
  <div
    className="placeholder"
    key="5"
  />
</section>
`;
